lib/resque/pool/config_throttle.rb

```diff
@@ -0,0 +1,35 @@
+module Resque
+  class Pool
+    # Throttle the frequency of loading pool configuration
+    class ConfigThrottle
+      def initialize(period, config_loader, time_source: Time)
+        @period = period
+        @config_loader = config_loader
+        @resettable = config_loader.respond_to?(:reset!)
+        @last_check = 0
+        @time_source = time_source
+      end
+
+      def call(env)
+        # We do not need to cache per `env`, since the value of `env` will not
+        # change during the life of the process.
```

Comment:
This comment (and my own playing around with decomposing the default loader) makes me feel that sending in `env` as an argument to `call` is the wrong API.  I'm currently thinking that `environment` should be passed in to the config loader's `initialize`, so we can codify the assumption that `env` will not change.